### PR TITLE
refactor(GUI): convert Progress Button to Rendition

### PR DESCRIPTION
**Pull request by amdomanska**
Convert progress-button component to Rendition

Signed-off-by: amdomanska <aga@resin.io>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
     "bluebird-retry": "0.11.0",
     "bootstrap-sass": "3.3.6",
     "chalk": "1.1.3",
+    "color": "https://registry.npmjs.org/color/-/color-2.0.1.tgz",
```
Comment (jhermsmeier) [inline package.json]: Looks like npm didn't quite do the right thing here – that should be

```json
"color": "2.0.1"
```

**Comment by jhermsmeier** [comment]
The `npm-shrinkwrap.json` hasn't been updated according to the changes in the `package.json` in here, did you `npm install --save --save-exact` to add the `color` dependency (that should automatically shrinkwrap for you and pin the version)?

File: .eslintrc.yml [hunk 2]
```diff
@@ -445,4 +445,3 @@ rules:
     - error
   node/no-extraneous-import:
     - error
-
```
[the remark this answers is not in the document]
Comment (amdomanska) [reply .eslintrc.yml]: Strange. Looks like (again) I've got some problems with my editor.

File: lib/gui/app/components/progress-button/progress-button.jsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+const React = require('react')
+const ReactDOM = require('react-dom')
+const propTypes = require('prop-types')
+const Color = require('color')
+
+import styled, { ThemeProvider } from 'styled-components'
+import { Button, ProgressBar, Provider } from 'rendition'
+
+import theme from './../../scss/modules/_theme'
```
Comment (amdomanska) [inline lib/gui/app/components/progress-button/progress-button.jsx]: Mix of require and imports to be unified

File: lib/gui/app/components/progress-button/progress-button.jsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  striped: propTypes.func,
+  active: propTypes.func,
+  percentage: propTypes.number,
+  label: propTypes.string,
+  shouldFlashStepBeDisabled: propTypes.func,
+  getLastFlashErrorCode: propTypes.func,
+  flash: propTypes.func,
+  image: propTypes.func,
+  device: propTypes.func,
+}
```
Comment (amdomanska) [inline lib/gui/app/components/progress-button/progress-button.jsx]: a few types should be changed

File: lib/gui/app/scss/modules/_theme.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  $paletteThemeDefaultForeground: '#b3b3b3',
+  $paletteThemePrimaryBackground: '#2297de',
+  $paletteThemePrimaryForeground: '#fff',
+  $paletteThemeWarningBackground: '#fca321',
+  $paletteThemeWarningForeground: '#fff',
+  $paletteThemeDangerBackground: '#d9534f',
+  $paletteThemeDangerForeground: '#fff',
+  $paletteThemeSuccessBackground: '#5fb835',
+  $paletteThemeSuccessForeground: '#fff'
+}
```
Comment (amdomanska) [inline lib/gui/app/scss/modules/_theme.js]: this file should be refactored the same as suggested in #2477

File: lib/gui/app/components/progress-button/progress-button.jsx (hunk 3 above)
Comment (amdomanska) [reply lib/gui/app/components/progress-button/progress-button.jsx]: remove _ prefix from _theme file

**Comment by lurch** [comment]
Eeek, looks like all the `"dev": true` entries have disappeared from the shrinkwrap?!
Are the instructions [here](https://github.com/resin-io/etcher/blob/master/docs/CONTRIBUTING.md#updating-a-dependency) still correct @jhermsmeier ?

**Comment by jhermsmeier** [comment]
@lurch yup, instructions are still correct (apart from that `--save-exact` is also used now)

**Comment by amdomanska** [comment]
@resinci retest

File: lib/gui/app/components/progress-button/progress-button.js [hunk 6]
```diff
@@ -16,13 +16,15 @@
 
 'use strict'
 
-/**
- * @module Etcher.Components.ProgressButton
- */
-
 const angular = require('angular')
+const { react2angular } = require('react2angular')
```
Comment (LucianBuzzo) [inline lib/gui/app/components/progress-button/progress-button.js]: Can you rename this file to `index.js` ?

File: lib/gui/app/pages/main/templates/main.tpl.html [hunk 7]
```diff
@@ -111,14 +111,17 @@
...
+          striped="main.state.getFlashState().type"
+          active = "main.state.isFlashing()"
           percentage="main.state.getFlashState().percentage"
-          striped="{{ main.state.getFlashState().type == 'check' }}"
-          ng-attr-active="{{ main.state.isFlashing() }}"
-          ng-click="flash.flashImageToDrive(main.selection.getImage(), main.selection.getSelectedDevices())"
-          ng-disabled="main.shouldFlashStepBeDisabled() || main.state.getLastFlashErrorCode()">
-            <span ng-bind="flash.getProgressButtonLabel()"></span>
+          label="flash.getProgressButtonLabel()"
+          should-flash-step-be-disabled="main.shouldFlashStepBeDisabled()"
```
Comment (LucianBuzzo) [inline lib/gui/app/pages/main/templates/main.tpl.html]: These two properties should just be merged together to create a single `disabled` property, like the old ng component had:
```
disabled="main.shouldFlashStepBeDisabled() || main.state.getLastFlashErrorCode()"
```

File: lib/gui/app/pages/main/templates/main.tpl.html [hunk 8]
```diff
@@ -111,14 +111,17 @@
       </div>
 
       <div class="space-vertical-large">
-        <progress-button class="button-brick"
+        <progress-button
           tabindex="3"
+          striped="main.state.getFlashState().type"
```
Comment (LucianBuzzo) [inline lib/gui/app/pages/main/templates/main.tpl.html]: This property should just be a boolean like it was previously

File: lib/gui/app/components/progress-button/progress-button.jsx [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+
+  background-size: 20px 20px;
+`
+
+class ProgressButton extends React.Component {
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
```
Comment (LucianBuzzo) [inline lib/gui/app/components/progress-button/progress-button.jsx]: This state seems completely unnecessary, can you just use `props` directly to determine these values?

File: lib/gui/app/components/progress-button/progress-button.jsx [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+  }
+
+  static getDerivedStateFromProps (props, state) {
+    state.disabled = props.shouldFlashStepBeDisabled || props.getLastFlashErrorCode
+    state.striped = props.striped == 'check'
+    return state
+  }
+
+  handleClick = () => {
+    this.setState({ openBar: true })
```
Comment (LucianBuzzo) [inline lib/gui/app/components/progress-button/progress-button.jsx]: The state key `openBar` is unused

File: lib/gui/app/components/progress-button/progress-button.jsx [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+    }
+  }
+
+  static getDerivedStateFromProps (props, state) {
+    state.disabled = props.shouldFlashStepBeDisabled || props.getLastFlashErrorCode
+    state.striped = props.striped == 'check'
+    return state
+  }
+
+  handleClick = () => {
```
Comment (LucianBuzzo) [inline lib/gui/app/components/progress-button/progress-button.jsx]: Is there no way to push the click handler upwards into the parent component like it was before?

**Pull request merged**